rav1e-by-gop/src/bin/encode.rs

```diff
@@ -157,13 +147,33 @@ pub fn perform_encode_inner<
         progress
     };
 
+    let num_local_slots = if num_threads == 0 {
+        0
+    } else {
+        let local_workers = opts.local_workers.unwrap_or_else(|| {
+            // workers tend to use about 56% of the available cpus
+            // tiles tend to use about the 40% of the available cpus
+            // letting rayon schedule the workload seems to deal with starvation
+            // reserve 1 thread for 4 parallel tiles
+
+            num_threads - opts.tiles * 2 / 8
```

Comment:
I'm a little unclear on the logic here. We use fewer workers the more tiles there are?

Reply:
The idea is share the pool across many workers but reduce the odds of hitting the situation in which the workloads are queuing up, that formula assumes we need to leave a thread for the tiling every 4 tiles. It is working fairly well for -s 6 and 20 hw threads, it will need tweaking and more testing though.

Reply:
Okay, I'll keep an eye on it but this seems fine for now.